ibis/backends/flink/registry.py

```diff
@@ -61,10 +67,11 @@ def _cast(translator: ExprTranslator, op: ops.generic.Cast) -> str:
         return f"CAST({arg_translated} AS date)"
     elif to.is_json():
         return arg_translated
-
-    from ibis.backends.base.sql.registry.main import cast
-
-    return cast(translator=translator, op=op)
+    elif op.arg.dtype.is_temporal() and op.to.is_int64():
+        return f"1000000 * unix_timestamp({arg_translated})"
+    else:
+        sql_type = type_to_sql_string(op.to)
+        return f"CAST({arg_translated} AS {sql_type})"
```

Comment:
What are these changes for? I'd need to double check (too many things change, so I could be misremembering), but I feel like I removed the very specific timestamp-to-int64 cast previously, because it was breaking some tests.

Reply:
This is exactly what `cast(translator=translator, op=op)` does. I just copied [the code from the function](https://github.com/ibis-project/ibis/blob/master/ibis/backends/base/sql/registry/main.py#L103C1-L110C54) instead of importing and using `cast`, because I need to override the implementation for `type_to_sql_string(op.to)`.